src/ast/passes/callback_visitor.h

```diff
@@ -2,50 +2,23 @@
 
 #include <functional>
 
-#include "ast/visitors.h"
+#include "ast/visitor.h"
 
 namespace bpftrace {
 namespace ast {
 
 using callback = std::function<void(Node *)>;
 
-class CallbackVisitor : public Visitor {
+class CallbackVisitor : public Visitor<CallbackVisitor> {
 public:
   explicit CallbackVisitor(ASTContext &ctx, callback func)
-      : Visitor(ctx), func_(func)
+      : Visitor<CallbackVisitor>(ctx), func_(func)
   {
   }
-
-  void visit(Integer &integer) override;
-  void visit(PositionalParameter &param) override;
-  void visit(String &string) override;
-  void visit(StackMode &mode) override;
-  void visit(Identifier &identifier) override;
-  void visit(Builtin &builtin) override;
-  void visit(Call &call) override;
-  void visit(Map &map) override;
-  void visit(Variable &var) override;
-  void visit(Binop &binop) override;
-  void visit(Unop &unop) override;
-  void visit(Ternary &ternary) override;
-  void visit(FieldAccess &acc) override;
-  void visit(ArrayAccess &arr) override;
-  void visit(Cast &cast) override;
-  void visit(Tuple &tuple) override;
-  void visit(ExprStatement &expr) override;
-  void visit(AssignMapStatement &assignment) override;
-  void visit(AssignVarStatement &assignment) override;
-  void visit(AssignConfigVarStatement &assignment) override;
-  void visit(If &if_node) override;
-  void visit(Unroll &unroll) override;
-  void visit(While &while_block) override;
-  void visit(Jump &jump) override;
-  void visit(Predicate &pred) override;
-  void visit(AttachPoint &ap) override;
-  void visit(Probe &probe) override;
-  void visit(Config &config) override;
-  void visit(Block &block) override;
-  void visit(Program &program) override;
+  void preVisit(Node &node)
```

Comment:
It seems like this is only really used by the node counting (unless I'm missing something). Should we be explicit with the naming and make it more generic in the future if needed?